src/test/java/com/example/moviemanager/controller/MovieRestControllerTests.java

```diff
@@ -0,0 +1,156 @@
+package com.example.moviemanager.controller;
+
+import com.example.moviemanager.entity.Movie;
+import com.example.moviemanager.service.MovieNotFoundException;
+import com.example.moviemanager.service.MovieService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest
+public class MovieRestControllerTests {
+
+    @MockBean
+    private MovieService movieService;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    public void returns_list_of_all_movies() throws Exception {
+        // given
+        List<Movie> movies = new ArrayList<>();
+        movies.add(new Movie(1, "Home Alone", "Christmas movie", 1990, 8.5));
+        movies.add(new Movie(2, "Home Alone 2", "Christmas movie", 1992, 8.9));
+
+        when(movieService.findAll()).thenReturn(movies);
+
+        // then
+        mockMvc.perform(get("/movies"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.size()").value(movies.size()))
+                .andExpect(content().string(objectMapper.writeValueAsString(movies)))
+                .andDo(print());
```

Comment:
I think it should be used only for debug purposes, but does not add real value now